packages/panzer/mini-em/src/eqn_sets/MiniEM_EquationSet_Maxwell_impl.hpp

```diff
@@ -0,0 +1,206 @@
+#ifndef _MiniEM_EquationSet_Maxwell_impl_hpp_
+#define _MiniEM_EquationSet_Maxwell_impl_hpp_
+
+
+#include "Teuchos_ParameterList.hpp"
+#include "Teuchos_StandardParameterEntryValidators.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Teuchos_Assert.hpp"
+#include "Phalanx_DataLayout_MDALayout.hpp"
+#include "Phalanx_FieldManager.hpp"
+
+#include "Panzer_IntegrationRule.hpp"
+#include "Panzer_BasisIRLayout.hpp"
+
+// include evaluators here
+#include "Panzer_Integrator_BasisTimesScalar.hpp"
+#include "Panzer_Integrator_TransientBasisTimesScalar.hpp"
+#include "Panzer_Integrator_BasisTimesVector.hpp"
+#include "Panzer_Integrator_CurlBasisDotVector.hpp"
+#include "Panzer_ScalarToVector.hpp"
+#include "Panzer_Sum.hpp"
+#include "Panzer_Constant.hpp"
+
+// ***********************************************************************
+template <typename EvalT>
+mini_em::EquationSet_Maxwell<EvalT>::
+EquationSet_Maxwell(const Teuchos::RCP<Teuchos::ParameterList>& params,
+    const int& default_integration_order,
+    const panzer::CellData& cell_data,
+    const Teuchos::RCP<panzer::GlobalData>& global_data,
+    const bool build_transient_support) :
+    panzer::EquationSet_DefaultImpl<EvalT>(params,default_integration_order,cell_data,global_data,build_transient_support ) {
+  // ********************
+  // Validate and parse parameter list
+  // ********************
+  {
+    Teuchos::ParameterList valid_parameters;
+    this->setDefaultValidParameters(valid_parameters);
+    /*  Equations are
+     *   1/eps dE/dt = 1/mu \nabla \times B -J
+     *   dB/dt = - \nabla \times E
+     *
+     *   in weak form
+     */
+
+
+    valid_parameters.set("Model ID","","Closure model id associated with this equation set");
+    valid_parameters.set("Basis Order",1,"Order of the basis");
+    valid_parameters.set("Integration Order",2,"Order of the integration");
+    valid_parameters.set("Epsilon",8.854187817e-12, "Permittivity of free space");
+    valid_parameters.set("Mu",1.2566370614e-6, "Permeability of free space");
+
+    params->validateParametersAndSetDefaults(valid_parameters);
+  }
+
+  int basis_order = params->get<int>("Basis Order");
+  std::string model_id = params->get<std::string>("Model ID");
+  int int_order = params->get<int>("Integration Order");
+  eps = params->get<double>("Epsilon");
+  mu = params->get<double>("Mu");
+  dimension = cell_data.baseCellDimension();
+  assert(dimension == 2 || dimension == 3);
```

Comment:
You use `TEUCHOS_ASSERT` in some places, and `assert` in others.  Inconsistency could mean unexpected behavior.  For example, `assert` goes away in release builds, while `TEUCHOS_ASSERT` may be controlled by separate Teuchos CMake options.